src/daft-parquet/src/file.rs

```diff
@@ -502,16 +502,31 @@ impl ParquetFileReader {
                             .into_iter()
                             .collect::<DaftResult<Vec<_>>>()?;
 
-                        let table_iter = arrow_column_iters_to_table_iter(
-                            arr_iters,
-                            row_range.start,
-                            daft_schema,
-                            uri,
-                            predicate,
-                            original_columns,
-                            original_num_rows,
-                        );
                         rayon::spawn(move || {
+                            // Even if there are no columns to read, we still need to create a empty table with the correct number of rows
+                            // This is because the columns may be present in other files. See https://github.com/Eventual-Inc/Daft/pull/2514
+                            if arr_iters.is_empty() {
+                                let table = Table::new_with_size(
+                                    Schema::empty(),
```

Comment:
I think this might be a departure in behavior from the current reader, where we return a schema-ful Table

But I don't think this is an issue since we do the cast later on anyways.